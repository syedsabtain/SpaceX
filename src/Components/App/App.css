.custom-nav a {
    color: white!important;
  text-shadow: 2px 2px 4px #000000!important;
}
.custom-nav button{
    background-color: white!important;
}
.custom-nav h2{
    color: white!important;
    text-shadow: 2px 2px 4px #000000!important;
    font-size: 29px;
    margin-left: 20px!important;
}
.bd
{
    border: 2px solid red;
}
.custom-nav span{
    color: white!important;
    text-shadow: 2px 2px 4px #000000!important;

}