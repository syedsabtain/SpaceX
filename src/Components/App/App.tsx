import React from 'react';
import List from '../launchlist/index'
import Launchprofile from '../launchdetails/index'
import {useState} from 'react';
import '../../bootstrap.min.css';
import './App.css';


function App() {
    let [id,
        setId] = useState(1);
    const handleIdChange = (newid : number) : void => {
        setId(newid);
    }
    return (
        <div className=" ">
            <nav
                className="navbar navbar-expand-lg     navbar-light   bg-transparent custom-nav">
                <h2 className="navbar-brand">SpaceX</h2>
                <button
                    className="navbar-toggler"
                    type="button"
                    data-toggle="collapse"
                    data-target="#navbarSupportedContent"
                    aria-controls="navbarSupportedContent"
                    aria-expanded="false"
                    aria-label="Toggle navigation">
                    <span className="navbar-toggler-icon"></span>
                </button>

                <div className="collapse navbar-collapse justify-content-center" id="navbarSupportedContent">
                    <ul className="navbar-nav mr-aut0  ">
                        
                        <li className='nav-item text-white ml-5'>
                            <span className='nav-link text-white'>Â© Created By Syed Sabtain</span>
                        </li>
                        <li className='nav-item ml-5'>
                           <a  href='https://github.com/syedsabtain' target='blank'><i className="fab fa-github fa-2x"></i></a>
                        </li>


                    </ul>
                    
                </div>
            </nav>

            <div className='container mr-0 ml-0'>
                <div className='row '>
                    <div className='col-md-3 '>
                   
                        <List handleIdChange={handleIdChange}></List>

                    </div>
                    <div className='col-md-9 justify-content-start '>
                        <Launchprofile id={id}></Launchprofile>

                    </div>
                </div>
            </div>

        </div>
    );
}

export default App;
