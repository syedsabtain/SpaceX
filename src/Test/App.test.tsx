import React from 'react';
import { shallow} from 'enzyme';
import App from '../Components/App/App';


describe('Testing App component Elements',()=>{

  it('Div',()=>{

    const component = shallow(<App></App>);

    expect(component.find('div').length).toEqual(6);

  });

  it('Button',()=>{
    const component = shallow(<App></App>);
    expect(component.find('button').length).toEqual(1);
  });

  it('ul and li',()=>{
    const component = shallow(<App></App>);
    expect(component.find('ul').length).toEqual(1);
    expect(component.find('li').length).toEqual(2)
  })
});

test('renders the component', () => {


  const component = shallow(<App />);
  expect(component).toMatchSnapshot();


});
