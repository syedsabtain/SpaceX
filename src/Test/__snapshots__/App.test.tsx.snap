// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<div
  className=" "
>
  <nav
    className="navbar navbar-expand-lg     navbar-light   bg-transparent custom-nav"
  >
    <h2
      className="navbar-brand"
    >
      SpaceX
    </h2>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarSupportedContent"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse justify-content-center"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav mr-aut0  "
      >
        <li
          className="nav-item text-white ml-5"
        >
          <span
            className="nav-link text-white"
          >
            Â© Created By Syed Sabtain
          </span>
        </li>
        <li
          className="nav-item ml-5"
        >
          <a
            href="https://github.com/syedsabtain"
            target="blank"
          >
            <i
              className="fab fa-github fa-2x"
            />
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    className="container mr-0 ml-0"
  >
    <div
      className="row "
    >
      <div
        className="col-md-3 "
      >
        <List
          handleIdChange={[Function]}
        />
      </div>
      <div
        className="col-md-9 justify-content-start "
      >
        <LaunchProfile
          id={1}
        />
      </div>
    </div>
  </div>
</div>
`;
